soroban-react-dapp/contracts/access_control/src/test.rs

```diff
@@ -0,0 +1,186 @@
+#![cfg(test)]
+
+extern crate std;
+
+use super::*;
+use soroban_sdk::{
+    symbol_short,
+    testutils::{Address as _, AuthorizedFunction, AuthorizedInvocation},
+    vec, Address, Env,
+};
+
+#[test]
+fn test_init() {
+    let env = Env::default();
+
+    let contract_id = env.register_contract(None, AccessControl);
+    let client = AccessControlClient::new(&env, &contract_id);
+
+    let admin = Address::generate(&env);
+
+    // Initialize the contract
+    let admin_title = String::from_str(&env, "admin_title");
+    client.init(&admin, &admin_title);
+
+    assert_eq!(client.get_admin(), admin);
+    assert_eq!(client.get_admin_title(), admin_title);
+}
+
+#[test]
+#[should_panic(expected = "contract already initilized")]
+fn test_init_again() {
+    let env = Env::default();
+
+    let contract_id = env.register_contract(None, AccessControl);
+    let client = AccessControlClient::new(&env, &contract_id);
+
+    let admin = Address::generate(&env);
+
+    // Initialize the contract
+    let admin_title = String::from_str(&env, "admin_title");
+    client.init(&admin, &admin_title);
+
+    // Initialize the contract again, should panic
+    client.init(&admin, &admin_title);
+}
+
+#[test]
+fn test_set_title_by_owner() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let contract_id = env.register_contract(None, AccessControl);
+    let client = AccessControlClient::new(&env, &contract_id);
+
+    let admin = Address::generate(&env);
+
+    // Initialize the contract
+    let admin_title = String::from_str(&env, "admin_title");
+    client.init(&admin, &admin_title);
+
+    // Check a new title
+    let new_admin_title = String::from_str(&env, "new_admin_title");
+    client.set_title(&new_admin_title);
+
+    assert_eq!(client.get_admin_title(), new_admin_title);
+}
+
+#[test]
+fn test_set_title_by_non_owner() {
+    let env = Env::default();
+
+    let contract_id = env.register_contract(None, AccessControl);
+    let client = AccessControlClient::new(&env, &contract_id);
+
+    let admin = Address::generate(&env);
+
+    // Initialize the contract
+    let admin_title = String::from_str(&env, "admin_title");
+    client.init(&admin, &admin_title);
+
+    // Check the authorization, should failed
+    let new_admin_title = String::from_str(&env, "new_admin_title");
+    let result = client.try_set_title(&new_admin_title);
+    assert!(result.is_err());
```

Comment:
this one is returning erros because auths where not mocked.
You need to choose, or do all tests mocking all auths or not